cpp/include/cuvs/core/nvtx.hpp

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
This shouldn't be a user-facing API and thus shouldn't be defined in `include/`. We should no longer be using `include/` for headers just because they need to be used across cuvs. Use `src/` for those headers. Only headers in `include/` should be those that we expect the user to interact with directly (thus the number of headers in `include/` should be very small.

Reply:
Having all of the declaration in `src/` will lead to `#include` with long relative paths that can be problematic in the future.
To avoid that we can add the `src/` directory to CMake `target_include_directories()` but this would lead to includes such as `src/core/nvtx.hpp`. Are we fine with that or do we want to keep something like `cuvs/core/nvtx.hpp`?

Reply:
I've done relative paths for cagra so far and haven't seen them to be too terrible. Is ivf-pq somehow making it more challenging to work with? 

From a development perspective, I generally tend to prefer the use of the relative quotationed paths for things that are local to src/, (and thus internal) rather than muddying the line between the two and making it harder to discern which things are public APIs and which aren't.